editor/src/clj/editor/lsp/server.clj

```diff
@@ -163,11 +163,29 @@
                          :else
                          (throw (ex-info "Invalid text document sync kind" {:value textDocumentSync})))})
 
-(def default-settings
-  {"Lua" {:diagnostics {:globals lua/defined-globals}}})
-
-(defn- configuration-handler [{:keys [items]}]
-  (mapv (comp default-settings :section) items))
+(defn- configuration-handler [project]
+  (fn [{:keys [items]}]
+    (lsp.async/with-auto-evaluation-context evaluation-context
+      (mapv
+        (fn [{:keys [section]}]
+          (case section
+            "Lua"
+            (let [script-intelligence (g/node-value project :script-intelligence evaluation-context)
+                  completions (g/node-value script-intelligence :lua-completions evaluation-context)]
+              {:diagnostics {:globals (into lua/defined-globals
+                                            (lua/extract-globals-from-completions completions))}})
+
+            "files.associations"
+            (let [workspace (g/node-value project :workspace evaluation-context)
+                  resource-types (g/node-value workspace :resource-types evaluation-context)]
+              (into {}
+                    (keep (fn [[ext {:keys [textual? language]}]]
```

Comment:
`:textual?` files include protobuf files such as `*.go` and `*.atlas` as well as settings files such as `game.project`. Do we want these to be included here? Or should we only include files that can be opened using the `:code` `:view`?

Reply:
Good question. So far, we only need information about files openable in the code view. When we will have workspace diagnostics though, it will be possible to use language servers to lint `.go` and `.atlas` files as a build step, but we currently don't define adequate language id for the protobuf text files (they are reported as `plaintext` while it should be something like [`textproto`](https://github.com/thejustinwalsh/textproto-grammar/blob/97efb452403712938d9e2165e4c003e4a0ef6ace/extension.js#L6) instead). Maybe we will allow editing these files as text in the future? Anyway, those files are excluded now because we don't report the fallback `plaintext` language.